agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -127,6 +130,9 @@ func TestTaskCleanupDoesNotDeadlock(t *testing.T) {
 		dockerId, err := agent.ResolveTaskDockerID(testTask, "1")
 		require.NoError(t, err, fmt.Sprintf("Cycle %d: Error resolving docker id for container in task", i))
 
+		err = testTask.Stop()
```

Comment:
The functional test was added to guard against regressions in task manager's `cleanupTask` method. Specifically to ensure that the task engine ignores/handles any task when it's cleaning a task. This should really be a unit/integ test and doesn't warrant a functional test IMO as it's hard to force that race condition to surface. Especially since the task def being used by the functional test is leading to false negatives most of the time.

In short, yes. @samuelkarp is right when he says 
>  I think that by making this change you're no longer exercising the codepath that was deadlocking.